internal/pkg/model/state.go

```diff
@@ -90,16 +154,48 @@ func (s *State) ConfigRows() (rows []*ConfigRowState) {
 	return rows
 }
 
-func (s *State) Get(key Key) ObjectState {
-	object, err := s.GetOrCreate(key)
-	if err != nil {
-		panic(err)
+func (s *State) ConfigRowsFrom(config ConfigKey) (rows []*ConfigRowState) {
+	for _, object := range s.All() {
+		if v, ok := object.(*ConfigRowState); ok {
+			if v.BranchId != config.BranchId || v.ComponentId != config.ComponentId || v.ConfigId != config.Id {
+				continue
+			}
+			rows = append(rows, v)
+		}
 	}
+	return rows
+}
 
-	if object == nil {
-		panic(fmt.Errorf(`object "%s" not found`, key.String()))
+// SearchForConfigRows by ID and name.
+func (s *State) SearchForConfigRows(str string, config ConfigKey) []*ConfigRowState {
+	matches := make([]*ConfigRowState, 0)
+	for _, object := range s.ConfigRowsFrom(config) {
+		if matchObjectIdOrName(str, object) {
+			matches = append(matches, object)
+		}
 	}
-	return object
+	return matches
+}
+
+// SearchForConfigRow by ID and name.
+func (s *State) SearchForConfigRow(str string, config ConfigKey) (*ConfigRowState, error) {
+	rows := s.SearchForConfigRows(str, config)
+	switch len(rows) {
+	case 1:
+		// ok, one match
+		return rows[0], nil
+	case 0:
+		return nil, fmt.Errorf(`no row matches the specified "%s"`, str)
+	default:
+		return nil, fmt.Errorf(`multiple rows match the specified "%s"`, str)
+	}
+}
+
+func (s *State) Get(key Key) ObjectState {
+	if v, ok := s.objects.Get(key.String()); ok {
+		return v.(ObjectState)
+	}
+	panic(fmt.Errorf(`%s not found`, key.Desc()))
```

Comment:
`Get` metoda sa vola, ked som si isty ze dany objekt existuje. No bola tam chyba a interne sa volalo `GetOrCreate`.
Pridal som na to aj test.